lib/data/models/currency.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:app/data/models/network_type.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'currency.freezed.dart';
+part 'currency.g.dart';
+
+@freezed
+class Currency with _$Currency {
+  const factory Currency({
+    required String currency,
```

Comment:
This is another place to use Currency imho. All string fields below indicates some numbers, and we shouldn't put them into strings.

Reply:
money2 is incompatible with app right now because of intl 18.0+

Reply:
Seems like it already in our app through transitive dependency (and we should add it as direct dependency along with money2improver). Seems like this comment is outdated?